evm-abci/src/bin/client.rs

```diff
@@ -1,83 +1,149 @@
 use ethers::prelude::*;
 use evm_abci::types::{Query, QueryResponse};
 use eyre::Result;
+use yansi::{Paint};
+use crate::User::*;
 
-#[tokio::main]
-async fn main() -> Result<()> {
-    // the ABCI port
-    let host = "http://127.0.0.1:3002";
+const ALICE: &str = "0xAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+const BOB: &str = "0xBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB";
+const CHARLIE: &str = "0xCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC";
 
-    let value = ethers::utils::parse_units(1, 18).unwrap();
-    let alice = "0x2c47Faeff6ED706E9C1D95f51F1800938Cf7c632".parse::<Address>()?;
-    let bob = "0xAAAAAAAAAAAAAABBBBBBBBBBBBBBBB938Cf7c632".parse::<Address>()?;
+#[derive(Copy, Clone)]
+enum User {
+    Alice,
+    Bob,
+    Charlie
+}
 
-    let client = reqwest::Client::new();
+fn user_to_address(user: User) -> Result<Address> {
+    match user {
+        Alice => Ok(ALICE.parse::<Address>()?),
+        Bob => Ok(BOB.parse::<Address>()?),
+        Charlie => Ok(CHARLIE.parse::<Address>()?),
+    }
+}
+
+fn user_to_name(user: User) -> &'static str {
+    match user {
+        Alice => "Alice",
+        Bob => "Bob",
+        Charlie => "Charlie",
+    }
+}
 
-    // Query balance before
-    let query = Query::Balance(alice);
-    let query = serde_json::to_string(&query).unwrap();
+fn get_readable_eth_value(value: U256) -> Result<f64> {
+    let value_string = ethers::utils::format_units(value, "ether")?;
+    Ok(value_string.parse::<f64>()?)
```

Comment:
this feels weird... i would think that `string.parse` returns a `Result`, so i could just return its result (without the `?` and wrapping that in an `Ok`), but when i try that i get a
```
note: expected enum `Result<_, ErrReport>`
              found enum `Result<_, ParseFloatError>`
```
and i thought `eyre::Result` would operate regardless of error type... but i must be missing something, and this approach is what the compiler recommended

Reply:
that's the idiomatic way to do it, you do `Ok(foo()?)` so that `?` converts the `ParseFloatError` of `.parse() ` to `eyre::ErrorReport` and returns it, and then you wrap it in an `Ok`